internal/certificates/certificates.go

```diff
@@ -0,0 +1,126 @@
+package certificates
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"os"
+	"sync"
+
+	"github.com/ydb-platform/ydb-go-sdk/v3/internal/xerrors"
+)
+
+var (
+	// fileCache
+	//  map[string][]*x509.Certificate
+	fileCache sync.Map
```

Comment:
Certificates never removed from cache, it can cause memory leak in long running applications.

I suggest to remove certificates from cache by timer/item count.

Reply:
The main problem is determining the suitable removal policy. For PEM cache the obvious solution is to remove expired (i.e. `time.Now() > cert.NotAfter`) certificates from cache. However, in applications with a lot of certificates with long expiration periods memory usage can still be big. For filename cache it is less obvious because the cached value is a slice of certs.

Implementing something smart like LRU/LFU seems to be an overkill in this situation, but I might be wrong here. We could also do something like FIFO with max cache records limit.

Need more info on real-world applications from which came the motivation for the issue.

Reply:
I think it will not be a big problem in real life, but the cache need any way to cleaning for prevent memory leak in rare cases.